tests/misra/test_misra.sh

```diff
@@ -36,7 +36,7 @@ echo "Cppcheck checkers list from test_misra.sh:" > $CHECKLIST
 
 cppcheck() {
   # get all gcc defines: arm-none-eabi-gcc -dM -E - < /dev/null
-  COMMON_DEFINES="-D__GNUC__=9 -UCMSIS_NVIC_VIRTUAL -UCMSIS_VECTAB_VIRTUAL"
+  COMMON_DEFINES="-D__GNUC__=9 -UCMSIS_NVIC_VIRTUAL -UCMSIS_VECTAB_VIRTUAL -DALLOW_DEBUG"
```

Comment:
what's this for? we want to test the release build

Reply:
cppcheck emits a false positive like[ this](https://github.com/commaai/panda/actions/runs/10859980022/job/30140003121#step:5:23) because of this [if statement](https://github.com/commaai/panda/blob/65f7852bb2362bc76c146a5bbac79dfcfedfaec4/board/safety/safety_ford.h#L412)

cppcheck does not understand the #ifdef ALLOW_DEBUG just above, which might change the variable.

I can add a suppression instead for that line only